### PR TITLE
Allow constant inputs to cost functions to be passed as floats

### LINKED ISSUE #38: Allow constant inputs to cost functions to be passed as floats
...and converted to `Variable` objects internally. 

This includes things like `dt` in GP cost function, and `cost_eps` or `cell_size` in `Collision2D`.

### PR DESCRIPTION
**Pull request by jeffin07**
## Motivation and Context
To close #38 

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by jeffin07** [comment]
@luisenp @mhmukadam I have added some  changes to `collisions.py` and  `double_integrator.py`. Tests not included. I will add test once the feature is finalized.Also i do see lot of `variables` in other cost functions,Could you please tell me which one are elligible for this change ?

File: theseus/embodied/collision/collision.py
Comment (mhmukadam) [inline theseus/embodied/collision/collision.py:37]: Likely need to add batch size here (and in rest of cost functions) same as pose.

File: theseus/embodied/motionmodel/double_integrator.py
Comment (luisenp) [inline theseus/embodied/motionmodel/double_integrator.py:36]: After the `ndim` check below, we can add `dt.data = self.dt.data.view(-1, 1)`, to make sure we have a batch dimension as well, as discussed above.

File: theseus/embodied/motionmodel/double_integrator.py
Comment (luisenp) [inline theseus/embodied/motionmodel/double_integrator.py:110]: I have added a fix for this class in #157, make sure you merge those changes as well.

File: theseus/embodied/motionmodel/double_integrator.py
Comment (luisenp) [inline theseus/embodied/motionmodel/double_integrator.py:196]: Similar comment about maintaining a batch shape here.

**Review by luisenp — CHANGES_REQUESTED** [review]
This is looking good! Some tweaks needed but mostly easy fixes and unit tests. The only other variable I've seen so far where this applies is `eff_radius` in `EffectorObjectContactPlanar`. Poses and sdf variables (origin, cell size, data) are multidimensional, so no need to change those.

File: theseus/embodied/collision/collision.py
Comment (mhmukadam) [reply theseus/embodied/collision/collision.py:37]: Thanks for clarifying Luis. @jeffin07, the batch size here you can get from the passed pose object.
Nice catch on the bug as well, Luis.

**Comment by jeffin07** [comment]
@luisenp Added the changes. Please review.Also do you have any suggestions for unit test ?

File: theseus/embodied/collision/collision.py
Comment (luisenp) [reply theseus/embodied/collision/collision.py:37]: It's probably better to use `(-1, 1)` instead of getting the batch size from the pose object, because `dt` is an element that we typically set a batch size of 1 for in our examples, regardless of the pose batch.

**Review by luisenp — CHANGES_REQUESTED** [review]
LGTM! For unit tests, just constructing all of these classes with the different types of inputs, and checking that the corresponding member is a variable of the expected type and shape should be sufficient. Thanks!!

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (luisenp) [inline theseus/embodied/collision/tests/test_eff_obj_contact.py:170]: Here we can check that `cost_function.eff_radius is eff_radius`.

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (luisenp) [inline theseus/embodied/collision/tests/test_eff_obj_contact.py:179]: Here we can add another check that `cost_function.eff_radius.data is eff_radius_t`.

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (luisenp) [inline theseus/embodied/collision/tests/test_eff_obj_contact.py:186]: Here we can first set up some random value for `eff_radius_f` and then add an additional check that `np.isclose(cost_function.eff_radius.data.item(), eff_radius_f)`.

File: theseus/embodied/motionmodel/tests/test_double_integrator.py
Comment (luisenp) [inline theseus/embodied/motionmodel/tests/test_double_integrator.py:64]: Add similar modifications to the test here, as mentioned above.

File: theseus/embodied/motionmodel/tests/test_double_integrator.py
Comment (luisenp) [inline theseus/embodied/motionmodel/tests/test_double_integrator.py:81]: We don't need the second check for `q_inv_v`.

**Review by luisenp — CHANGES_REQUESTED** [review]
Looks good, almost there! Once the code is finished, I'll also run some tests on my side before confirming that we can merge. Thanks!

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (jeffin07) [reply theseus/embodied/collision/tests/test_eff_obj_contact.py:186]: @luisenp 
we're already giving a random value to `eff_radius_f=1.0`
So  is this enough ` assert np.isclose(cost_function.eff_radius.data.item(), eff_radius_f)`

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (jeffin07) [reply theseus/embodied/collision/tests/test_eff_obj_contact.py:179]: @luisenp While `assert cost_function.eff_radius.data is eff_radius_t` gives error as one is `tensor` and other is `th.Variable` .Is it better to use  `assert torch.allclose(cost_function.eff_radius.data,eff_radius_t)` ?

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (luisenp) [reply theseus/embodied/collision/tests/test_eff_obj_contact.py:186]: By random I mean sampling a new value every iteration rather than hardcoding to 1.0.

**Comment by jeffin07** [comment]
@luisenp Added changes please review

**Comment by jeffin07** [comment]
@luisenp @mhmukadam I'm unable to run test after fetching latest chainges from `upstream/main`. When running `pytest theseus/` gives the following output. Any idea what I've done wrong ?

```
ImportError while importing test module '/theseus/theseus/optimizer/tests/test_sparse_linearization.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.7/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
theseus/optimizer/tests/test_sparse_linearization.py:9: in <module>
    import theseus as th
theseus/__init__.py:71: in <module>
    import theseus.embodied as eb
theseus/embodied/__init__.py:6: in <module>
    from .collision import Collision2D, EffectorObjectContactPlanar, SignedDistanceField2D
theseus/embodied/collision/__init__.py:6: in <module>
    from .collision import Collision2D
theseus/embodied/collision/collision.py:11: in <module>
    from theseus.embodied.kinematics import IdentityModel, KinematicsModel
theseus/embodied/kinematics/__init__.py:6: in <module>
    from .kinematics_model import IdentityModel, KinematicsModel, UrdfRobotModel
theseus/embodied/kinematics/kinematics_model.py:9: in <module>
    import differentiable_robot_model as drm
E   ModuleNotFoundError: No module named 'differentiable_robot_model'





```

This keeps repating for every file

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (jeffin07) [reply theseus/embodied/collision/tests/test_eff_obj_contact.py:181]: @luisenp when i try to do `assert cost_function.eff_radius.data is eff_radius_t` i get this error which expained above 

```
>           assert cost_function.eff_radius.data is eff_radius_t
E           assert tensor([[0.4160]], dtype=torch.float64) is tensor([[0.4160]], dtype=torch.float64)
E            +  where tensor([[0.4160]], dtype=torch.float64) = Variable(data=tensor([[0.4160]], dtype=torch.float64), name=Variable__680).data
E            +    where Variable(data=tensor([[0.4160]], dtype=torch.float64), name=Variable__680) = <theseus.embodied.collision.eff_obj_contact.EffectorObjectContactPlanar object at 0x7f813704edd0>.eff_radius

theseus/embodied/collision/tests/test_eff_obj_contact.py:180: AssertionError


```

Comment (luisenp) [reply theseus/embodied/collision/tests/test_eff_obj_contact.py:181]: Ah, right, it's because of the `view(-1, 1)` we added. In this case, let's stick to compare that they are close in value as you had initially.

Comment (jeffin07) [reply theseus/embodied/collision/tests/test_eff_obj_contact.py:181]: @luisenp Then can you take a look at PR. I've already committed the changes

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (luisenp) [reply theseus/embodied/collision/tests/test_eff_obj_contact.py:170]: You can delete the line above that has the `isinstance` check now.

File: theseus/embodied/motionmodel/tests/test_double_integrator.py
Comment (luisenp) [inline theseus/embodied/motionmodel/tests/test_double_integrator.py:87]: Sorry, didn't think of this before, but we should also check that `cost_weight.dt` has a batch dimension. And similar for `eff_radius_t` in the other cost function. We are also missing the check `np.isclose(cost_function.eff_radius.data.item(), eff_radius_f)` (and similar for `dt`).

**Comment by jeffin07** [comment]
@luisenp Updated unit tests

File: theseus/embodied/motionmodel/tests/test_double_integrator.py
Comment (luisenp) [reply theseus/embodied/motionmodel/tests/test_double_integrator.py:88]: @jeffin07 Missing the `np.allclose()` check for `eff_radius_f` and also checking that the batch dimension is present.

**Comment by jeffin07** [comment]
@luisenp Updated the changes.Please review

**Review by luisenp — APPROVED** [review]
Sorry for the delay, got busy with other PRs and forgot this was waiting for review. LGTM!

**Comment by luisenp** [comment]
Thanks a lot for another great contribution, @jeffin07 :)

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (mhmukadam) [inline theseus/embodied/collision/tests/test_eff_obj_contact.py:11]: Don't need this since it is already available as `th.Variable`

**Review by mhmukadam — APPROVED** [review]
Looks good! Thanks.

File: theseus/embodied/collision/tests/test_eff_obj_contact.py
Comment (jeffin07) [reply theseus/embodied/collision/tests/test_eff_obj_contact.py:11]: @mhmukadam Do i need to change this ?

**Comment by jeffin07** [comment]
@luisenp @mhmukadam Updated the changes. Please review

**Pull request merged**
